### DISCUSSION TITLE
Shorthand for simple effect propagating functions

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#8730] [Ideas]
```
def applyTwiceOnArray(f: t -> t \ ef1, g: t -> t \ ef2, arr: Array[t, r]): Array[t, r] \ ef1 + ef2 + r = ???
```
Introduce the shorthand
```
<keyword> def applyTwiceOnArray(f: t -> t, g: t -> t, arr: Array[t, r]): Array[t, r] = ???
```
That
* Adds effects to arrows
* Collects *syntactic* type variables of kind `Eff` in the signature
* Uses the union of these two sets as the def effect

This is limited, certainly, but avoid verbosity and cutting for simple effect propagating functions

**Comment by magnus-madsen** [comment]
This is idea has been proposed in the research literature.

**Reply by JonathanStarup** [reply]
I was inspired by Brachthäuser's talk we watched today. I think it could be really helpful to all levels of programmers actually if we find a good keyword
